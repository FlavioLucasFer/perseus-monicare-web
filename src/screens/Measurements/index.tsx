import React, { useEffect, useState } from 'react';
import { Timeline, Tooltip, Typography } from 'antd';
import moment from 'moment';
import { useParams } from 'react-router-dom';

import { 
  Patient, 
  PatientMeasurement, 
  PatientRepository 
} from 'api';

const { Item } = Timeline;
const { Title } = Typography;


const Measurements = () => {
  const [patient, setPatient] = useState<Patient | undefined>();
  const [measurements, setMeasurements] = useState<PatientMeasurement[]>([]);

  
  const { patientId } = useParams();
  
  useEffect(() => {
    const patientRepository = new PatientRepository();
    
    async function fetchMeasurements() {
      const patient = await patientRepository.findById(Number(patientId));
      const measurements = await patientRepository.getMeasurements(Number(patientId));
      
      setPatient(patient);
      setMeasurements(measurements);
    }

    fetchMeasurements();
  }, []);
  
  const renderTimeLineItem = (e: PatientMeasurement) => {    
    const {
      id,
      value,
      measuredAt,
      status,
      measurementType,
    } = e;
    
    let color = '#33f587';
    let dotTooltip = 'Tudo bem!';

    switch (status) {
      case 'bad':
        color = '#f53933';
        dotTooltip = 'Perigoso!'
        break;

      case 'caution':
        color = '#ffcc00';
        dotTooltip = 'Cuidado!'
        break;
    }

    return (
      <Item 
        key={id}
        label={<span className="fs-7"><b>{measurementType?.name}</b></span>}
        className="mt-2 mb-2"
        dot={
          <Tooltip title={dotTooltip}>
            <div style={{
              height: '15px',
              width: '15px',
              backgroundColor: 'transparent',
              border: `3px solid ${color}`,
              borderRadius: '50%',
             }} />
          </Tooltip>
        }>
        <span className={`fs-5 ${status === 'bad' && 'fw-bold'}`}
          style={{ color: status === 'bad' ? color : '#000' }}> 
          {value} 
        </span>
        <br />
        <small> <b> Medido em: </b> {moment(measuredAt).format('DD/MM/YYYY')} </small>
      </Item>
    );
  }

  const renderTimeLine = () => {
    return (
      <Timeline mode='left'
        reverse
        style={{
          overflowY: 'auto',
          overflowX: 'hidden',
          height: '100%',
        }}>
        {measurements.map(renderTimeLineItem)}
      </Timeline>
    );
  }

  return (
    <div className='container' style={{ height: '100%' }}>
      <div className='row col-sm-12' style={{ height: '100%' }}>
        <Title 
          className='text-center mb-2'
          level={2}>
          {patient?.name}
        </Title>

        <div className='col-sm-6' style={{ height: '85%' }}>
          {renderTimeLine()}
        </div>
      </div>
    </div>
  );
}

export default Measurements;
