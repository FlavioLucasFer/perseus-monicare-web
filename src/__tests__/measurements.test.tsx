import React from 'react';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Router } from 'react-router-dom';
import Measurements from 'screens/Measurements';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom') as any, 
	useNavigate: () => mockedUseNavigate,
}));

describe('Measurements Screen', () => {
	it('Should render properly', () => {
		const wrapper = shallow(
			<Measurements />
		);
		expect(wrapper.debug()).toMatchSnapshot();

	}); 

});