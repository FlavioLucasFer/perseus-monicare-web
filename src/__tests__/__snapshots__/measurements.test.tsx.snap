// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Measurements Screen Should render properly 1`] = `
"<div className=\\"container\\" style={{...}}>
  <div className=\\"row col-sm-12\\" style={{...}}>
    <Title className=\\"text-center mb-2\\" level={2} />
    <div className=\\"col-sm-6\\" style={{...}}>
      <Timeline mode=\\"left\\" reverse={true} style={{...}} />
    </div>
  </div>
</div>"
`;
